WORKSPACE

```diff
@@ -111,3 +111,21 @@ http_file(
     executable = True,
     sha256 = "dd8d58429a258b094b20a1435be3086ecee5d036b87c0e766309842766bc345b"
 )
+
+http_archive(
```

Comment:
I'm wondering - what's the standard equivalent of Maven (package management) in cpp?
If 2 dependencies both depend on dep A, but with different versions, how is that resolved?

I see that Abseil recommend "living at HEAD", which I like, but not everybody does that :)
https://abseil.io/about/philosophy#we-recommend-that-you-choose-to-live-at-head

Reply:
There is no common package management for C++. Bazel tries to do its own, the Buck build system has Buckaroo, Visual Studio has vcpkg. There's been a recent boom in C++ package managers in the past year or so, but it's too early for there to be any clear winners or losers yet.

Diamond dependencies are yet another unsolved problem. I know some library projects will go out of their way to rewrite things instead of using libraries, so that programs which use them are less likely to have a diamond dependency problem. Barring that, I would hope the libraries are written in such a way that two copies can coexist (no program-scope globals, no global constructors), barring that I would hope that updating the codebase that depends on the older one to use the newer one isn't that hard. For a much longer discussion, see Titus Winters' talk at cppcon, https://www.youtube.com/watch?v=tISy7EJQPzI&t=15m0s when announcing Abseil.

Reply:
:-)

Reply:
>Visual Studio has vcpkg

Dev on vcpkg here! We hope Vcpkg is useful to all C++ developers on all platforms, not just Visual Studio users. We recently expanded to support Linux and OSX and we know others have successfully used us to target more exotic platforms like Android, iOS, and Emscripten.

If you have any feedback for us about something we could do to make vcpkg more useful for you, I'd love to hear it!

Reply:
Hey neighbor! Haven't tried it and I don't do much C++ development these days.
Thanks for dropping by!

Reply:
Done!

ras0219-msft, sorry to suggest that vcpkg was tied to Visual Studio! I know it isn't, what I meant to say was that it came from the VC++ team as that team's solution to C++ packaging, not that it's tied to Visual Studio. FWIW, I've heard good things about vcpkg!